### PR TITLE
chore(deps): upgrade mongoose

### PR DESCRIPTION
**Pull request by bodymindarts**

### CODE REVIEW

**Review by krtk6160 — APPROVED** [review]

File: src/services/mongoose/accounts.ts [hunk 1]
```diff
@@ -62,8 +62,8 @@ export const AccountsRepository = (): IAccountsRepository => {
     try {
       const accounts: UserRecord[] = await User.find(
         {
-          title: { $exists: true, $ne: null },
-          coordinates: { $exists: true, $ne: null },
+          title: { $exists: true, $ne: undefined },
```
Comment (dolcalmi) [inline src/services/mongoose/accounts.ts:65]: did we do a migration forr this?

File: yarn.lock [hunk 2]
```diff
@@ -1868,6 +1868,13 @@
   dependencies:
     bson "*"
 
+"@types/bson@1.x || 4.0.x":
```
Comment (dolcalmi) [inline yarn.lock:1871]: please remove and re-create yarn.lock with `yarn install`

File: src/services/mongoose/accounts.ts (hunk 1 above)
Comment (bodymindarts) [reply src/services/mongoose/accounts.ts:65]: no - migration for this... I guess we should check if this is an issue

File: src/graphql/root/mutation/device-notification-token-create.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ const DeviceNotificationTokenCreateMutation = GT.Field<
       const user = await User.findOne({ _id: toObjectId<UserId>(uid) })
       if (!user) throw Error("find user issue")
 
-      user.deviceToken.addToSet(deviceToken)
+      user.deviceToken.push(deviceToken)
```
Comment (dolcalmi) [inline src/graphql/root/mutation/device-notification-token-create.ts:31]: if you change this you need to validate that device token is not present in the array

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: src/graphql/root/mutation/device-notification-token-create.ts (hunk 3 above)
Comment (bodymindarts) [reply src/graphql/root/mutation/device-notification-token-create.ts:31]: I see... perhaps there is another method `addToSet` is no longer present.

File: src/services/mongoose/accounts.ts (hunk 1 above)
Comment (bodymindarts) [reply src/services/mongoose/accounts.ts:65]: I have checked on bbw that the queries:
```
{ title: { $exists: true, $ne: null }, coordinates: { $exists: true, $ne: null} }
```
and 
```
{ title: { $exists: true, $ne: undefined }, coordinates: { $exists: true, $ne: undefined } }
```
yield identical results so I'd say we do not need a migration for this.

**Review by krtk6160 — APPROVED** [review]

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**